This is commit 762bcb8c50fd0256a545ad6ed2304403daca8081 from
Markus Steinborn <gnugv_maintainer@yahoo.de> Xaw3d git repo, with
the making a missing fontset abort changes removed, while keeping the
checks for missing fonts as those can lead to NULL ptr dereferences.

The checks for missing font sets is removed, because when the locale
is set to a UTF-8 locale (default now a days on most Linux distributions),
libXt will usually print:

Warning: Missing charsets in String to FontSet conversion
Warning: Unable to load any usable fontset

When resolving any fontset string, and return a NULL fontset.
For more details on this issue see:
http://tracker.xemacs.org/XEmacs/its/issue650

This however is not a fatal error (it means the default font set will be
used, and that some characters of the entire unicode range may not render
properly).

The changes in the original commit made this a fatal error causing many
programs, which worked fine before, to abort.

Below is the header of the orignal patch for reference:

From 762bcb8c50fd0256a545ad6ed2304403daca8081 Mon Sep 17 00:00:00 2001
From: Markus Steinborn <gnugv_maintainer@yahoo.de>
Date: Sat, 24 Apr 2010 16:54:35 +0200
Subject: [PATCH] Xaw3d-1.5E-fontset.patch

---
 xc/lib/Xaw3d/AsciiSink.c |    2 ++
 xc/lib/Xaw3d/Command.c   |    2 ++
 xc/lib/Xaw3d/Label.c     |   11 ++++++++++-
 xc/lib/Xaw3d/List.c      |   14 ++++++++++++--
 xc/lib/Xaw3d/MultiSink.c |   34 ++++++++++++++++++++++++++--------
 xc/lib/Xaw3d/SmeBSB.c    |    2 ++
 xc/lib/Xaw3d/Tip.c       |   16 ++++++++++++++--
 xc/lib/Xaw3d/XawIm.c     |   14 +++++++++-----
 8 files changed, 77 insertions(+), 18 deletions(-)

diff --git a/xc/lib/Xaw3d/AsciiSink.c b/xc/lib/Xaw3d/AsciiSink.c
index 2ebd952..a664db5 100644
--- a/xc/lib/Xaw3d/AsciiSink.c
+++ b/xc/lib/Xaw3d/AsciiSink.c
@@ -532,6 +532,8 @@ Cardinal *num_args;
 {
     AsciiSinkObject sink = (AsciiSinkObject) new;
 
+    if (!sink->ascii_sink.font) XtError("Aborting: no font found\n");
+
     GetGC(sink);
     
     sink->ascii_sink.insertCursorOn= CreateInsertCursor(XtScreenOfObject(new));
diff --git a/xc/lib/Xaw3d/Command.c b/xc/lib/Xaw3d/Command.c
index 6e04860..9f22d78 100644
--- a/xc/lib/Xaw3d/Command.c
+++ b/xc/lib/Xaw3d/Command.c
@@ -215,6 +215,8 @@ Cardinal *num_args;		/* unused */
   CommandWidget cbw = (CommandWidget) new;
   int shape_event_base, shape_error_base;
 
+  if (!cbw->label.font) XtError("Aborting: no font found\n");
+
   if (cbw->command.shape_style != XawShapeRectangle
       && !XShapeQueryExtension(XtDisplay(new), &shape_event_base, 
 			       &shape_error_base))
diff --git a/xc/lib/Xaw3d/List.c b/xc/lib/Xaw3d/List.c
index 99faf31..e7d7d57 100644
--- a/xc/lib/Xaw3d/List.c
+++ b/xc/lib/Xaw3d/List.c
@@ -366,6 +366,8 @@ Cardinal *num_args;
 {
     ListWidget lw = (ListWidget) new;
 
+    if (!lw->list.font) XtError("Aborting: no font found\n");
+
 /* 
  * Initialize all private resources.
  */
diff --git a/xc/lib/Xaw3d/SmeBSB.c b/xc/lib/Xaw3d/SmeBSB.c
index abf811f..dbb39aa 100644
--- a/xc/lib/Xaw3d/SmeBSB.c
+++ b/xc/lib/Xaw3d/SmeBSB.c
@@ -203,6 +203,8 @@ Cardinal *num_args;
     else
 	entry->sme_bsb.label = XtNewString( entry->sme_bsb.label );
 
+    if (!entry->sme_bsb.font) XtError("Aborting: no font found\n");
+
     CreateGCs(new);
 
     GetBitmapInfo(new, TRUE);	/* Left Bitmap Info */
diff --git a/xc/lib/Xaw3d/Tip.c b/xc/lib/Xaw3d/Tip.c
index 888eeac..233cc8d 100644
--- a/xc/lib/Xaw3d/Tip.c
+++ b/xc/lib/Xaw3d/Tip.c
@@ -252,6 +252,8 @@ Cardinal *num_args;
     TipWidget tip = (TipWidget)w;
     XGCValues values;
 
+    if (!tip->tip.font) XtError("Aborting: no font found\n");
+
     tip->tip.timer = 0;
 
     values.foreground = tip->tip.foreground;
